test/metabase/pulse/render/body_test.clj

```diff
@@ -414,28 +415,28 @@
     (is (has-inline-image?
          (render-bar-graph {:cols         default-columns
                             :rows         [[10.0 1] [5.0 10] [2.50 20] [1.25 30]]
-                            :viz-settings {}}))))
+                            :viz-settings {:graph.metrics ["NumPurchased"]}}))))
```

Comment:
Curious why these all needed to be updated?

Reply:
In a real question, if viz. has been set for line, area, or bar, the :viz-settings key will have :graph.metrics and :graph.dimensions values. 

The axis grouping strategy requires those keys to be able to actually re-construct the series. For example, a row `[timestamp 1 2 3 4 5]` needs to know that each of those values corresponds to a metric.